build-tools/packages/build-cli/src/commands/release/setDtsTypes.ts

```diff
@@ -0,0 +1,154 @@
+
+interface PackageTypesList {
+	packagesUpdated: string[];
+	packagesNotUpdated: string[];
+}
+
+interface UpdateConfig {
+	pkg: Package;
+	releaseType: string;
+	json: PackageJson;
+	typesFolder: string;
+	includesAlphaBeta: boolean;
+}
+
+const knownReleaseTag = ["alpha", "beta", "public"] as const;
+
+type ReleaseTag = typeof knownReleaseTag[number];
+
+function isReleaseTag(str: string | undefined): str is ReleaseTag {
+	return str === undefined ? false : knownReleaseTag.includes(str as any);
+}
+
+export default class SetReleaseTagPublishingCommand extends PackageCommand<
+	typeof SetReleaseTagPublishingCommand
+> {
+	static readonly description =
+		"Updates the .d.ts types that are included in the `types` field in package.json. This command is intended to be used in CI only.";
+
+	static readonly enableJsonFlag = true;
+
+	static readonly flags = {
+		types: Flags.custom<ReleaseTag>({
+			description: "Which .d.ts types to include in the published package.",
+			parse: async (input) => {
+				if (isReleaseTag(input)) {
+					return input;
+				}
+				throw new Error(`Invalid release type: ${input}`);
+			},
+			required: true,
+		})(),
+		...PackageCommand.flags,
+	};
+
+	private readonly packageList: PackageTypesList;
+
+	constructor(argv: string[], config: Config) {
+		super(argv, config);
+		this.packageList = {
+			packagesNotUpdated: [],
+			packagesUpdated: [],
+		};
+	}
+
+	protected async processPackage(pkg: Package): Promise<void> {
+		let packageUpdate: boolean = false;
+
+		updatePackageJsonFile(pkg.directory, (json) => {
+			const types: string | undefined = json.types ?? json.typings;
+
+			if (types === undefined) {
+				return;
+			}
+
+			const [typesFolder] = types.split("/");
+			const includesAlphaBeta: boolean = types.includes("alpha") || types.includes("beta");
+
+			const config: UpdateConfig = {
+				pkg,
+				releaseType: this.flags.types as ReleaseTag,
+				json,
+				typesFolder,
+				includesAlphaBeta,
+			};
+
+			packageUpdate = updatePackageJsonTypes(config, this.logger);
+
+			updatePackageLists(this.packageList, packageUpdate, pkg);
+		});
+	}
+
+	public async run(): Promise<PackageTypesList> {
+		// Calls processPackage on all packages.
+		await super.run();
+
+		const flags = this.flags;
+
+		if (this.packageList.packagesUpdated.length === 0) {
+			this.log(`No updates in package.json for ${flags.types} release tag`);
+		}
+
+		return this.packageList;
+	}
+}
+
+function updatePackageJsonTypes(config: UpdateConfig, log: CommandLogger): boolean {
+	const apiExtractorConfigFilePath = path.join(config.pkg.directory, "api-extractor.json");
+	let loadFile;
+	try {
+		loadFile = ExtractorConfig.loadFile(apiExtractorConfigFilePath);
+		const dtsRollupEnabled = loadFile.dtsRollup?.enabled;
+		if (dtsRollupEnabled !== undefined && dtsRollupEnabled) {
+			log.log(`Config exists: ${JSON.stringify(config.pkg.directory)}`);
+
+			const alpha = loadFile.dtsRollup?.alphaTrimmedFilePath;
```

Comment:
How have you been testing this? I think this may return a value even when one is not configured explicitly.

Reply:
yeah. that's correct. probably, removing `betaTrimmedFilePath` from `build-common`: https://github.com/microsoft/FluidFramework/blob/main/common/build/build-common/api-extractor-common.json#L43 would fix this

Reply:
Only SharedTree2 has dtsRollup enabled: https://github.com/microsoft/FluidFramework/blob/main/experimental/dds/tree2/api-extractor.json#L8. 

`build-common` does not have `alphaTrimmedFilePath` configured: https://github.com/microsoft/FluidFramework/blob/main/common/build/build-common/api-extractor-common.json#L43

I modified the tree2 api-extractor config, replacing `alphaTrimmedFilePath` with `betaTrimmedFilePath`. 

Later, 
`./bin/dev release setDtsTypes -g client --types alpha` didn't update the alpha types.  

so probably removing the betaTrimmedFilePath or rather dtsRollup from the common config should resolve this.

Reply:
PR removing rollup configs: https://github.com/microsoft/FluidFramework/pull/17502

Reply:
I don't think that is the correct change. With that change there is no longer a default value for those fields I think. When you run this for a package, what are the api-extractor settings that get loaded? Are you sure that it is loading the config and inheriting the settings from the base config?

Reply:
yeah I think so. I tested this by logging the alpha/beta file paths for tree2.  

`<projectFolder>/dist/<unscopedPackageName>.d.ts`
`<projectFolder>/dist/<unscopedPackageName>-beta.d.ts`

For any omitted fields, the API Extractor default values are merged.

Reply:
> I updated the api-extractor config for tree2 to have this dtsRollup section
> 
> ```json
> 	"dtsRollup": {
> 		"enabled": true,
> 		"untrimmedFilePath": "<projectFolder>/dist/<unscopedPackageName>.d.ts",
>     "alphaTrimmedFilePath": "<projectFolder>/dist/<unscopedPackageName>-alpha.d.ts",
> 		"betaTrimmedFilePath": "<projectFolder>/dist/<unscopedPackageName>-beta.d.ts",
> 		"publicTrimmedFilePath": "<projectFolder>/dist/<unscopedPackageName>-public.d.ts"
> 	},
> ```
> 
> After running build:docs, the alpha and beta types were generated. But when I ran `flub release setDtsTypes --dir . -v --types alpha`, nothing happened.

hmm...works fine with the release group flag (flub release setDtsTypes -g client -v --types alpha)